reference-architecture/azure-ansible/bastion.sh

```diff
@@ -0,0 +1,345 @@
+openshift_router_selector='region=infra'
+openshift_registry_selector='region=infra'
+
+ansible_become=yes
+ansible_ssh_user=${AUSERNAME}
+remote_user=${AUSERNAME}
+
+openshift_master_default_subdomain=${ROUTEREXTIP}.xip.io 
+openshift_use_dnsmasq=False
+openshift_public_hostname=${RESOURCEGROUP}.trafficmanager.net
+
+openshift_master_cluster_method=native
+openshift_master_cluster_hostname=${RESOURCEGROUP}.trafficmanager.net
+openshift_master_cluster_public_hostname=${RESOURCEGROUP}.trafficmanager.net
+
+# Enable cockpit
+osm_use_cockpit=true
+
+# Set cockpit plugins
+osm_cockpit_plugins=['cockpit-kubernetes']
+
+# default storage plugin dependencies to install, by default the ceph and
+# glusterfs plugin dependencies will be installed, if available.
+osn_storage_plugin_deps=['iscsi']
+
+[masters]
+master1 openshift_node_labels="{'role': 'master'}"
+master2 openshift_node_labels="{'role': 'master'}"
+master3 openshift_node_labels="{'role': 'master'}"
+
+[etcd]
+master1
+master2
+master3
+
+[nodes]
+master1 openshift_node_labels="{'region':'master','zone':'default'}" 
+master2 openshift_node_labels="{'region':'master','zone':'default'}" 
+master3 openshift_node_labels="{'region':'master','zone':'default'}" 
+node[01:${NODECOUNT}] openshift_node_labels="{'region': 'primary', 'zone': 'default'}"
+infranode1 openshift_node_labels="{'region': 'infra', 'zone': 'default'}"
+infranode2 openshift_node_labels="{'region': 'infra', 'zone': 'default'}"
+infranode3 openshift_node_labels="{'region': 'infra', 'zone': 'default'}"
+
+[quotanodes]
+master1 openshift_node_labels="{'region':'master','zone':'default'}" 
+master2 openshift_node_labels="{'region':'master','zone':'default'}" 
+master3 openshift_node_labels="{'region':'master','zone':'default'}" 
+node[01:${NODECOUNT}] openshift_node_labels="{'region': 'primary', 'zone': 'default'}"
+
+[misc]
+store1
+EOF
+
+
+cat <<EOF > /home/${AUSERNAME}/subscribe.yml
+---
+- hosts: all
+  vars:
+    description: "Subscribe OSE"
+  tasks:
+  - name: wait for .updateok
+    wait_for: path=/root/.updateok
+  - name: check connection
+    ping:
+  - name: Get rid of rhui repos
+    file: path=/etc/yum.repos.d/rh-cloud.repo state=absent
+  - name: Get rid of rhui Load balancers
+    file: path=/etc/yum.repos.d/rhui-load-balancers state=absent
+  - name: remove the RHUI package
+    yum: name=RHEL7 state=absent
+  - name: Get rid of old subs
+    shell: subscription-manager unregister
+    ignore_errors: yes
+  - name: register hosts
+    shell: subscription-manager register --username ${RHNUSERNAME} --password ${RHNPASSWORD} 
+    register: task_result
+    until: task_result.rc == 0
+    retries: 10
+    delay: 30
+    ignore_errors: yes
+  - name: attach sub
+    shell: subscription-manager attach --pool=$RHNPOOLID
+    register: task_result
+    until: task_result.rc == 0
+    retries: 10
+    delay: 30
+    ignore_errors: yes
+  - name: disable all repos
+    shell: subscription-manager repos --disable="*" 
```

Comment:
Guys they dont work. subscription-manager ansible module is unreliable. This is done this way for reliablity based on 200+ runs. Even disable * does not work. Its wonderful to be "clean", but reality is to get things to run everytime, with ansible its the way it is. Changing it to the common role, is where i started from, and wasted lots of time. 
Changing this will "look" cleaner and fail often.